aws/src/main/java/org/apache/iceberg/aws/s3/S3InputStream.java

```diff
@@ -194,9 +195,29 @@ private void openStream() throws IOException {
     }
   }
 
-  private void closeStream() throws IOException {
+  private void closeStream() {
     if (stream != null) {
-      stream.close();
+      // if we aren't at the end of the stream, and the stream is abortable, then
+      // call abort() so we don't read the remaining data with the Apache HTTP client
+      abortStream();
+      try {
+        stream.close();
+      } catch (Exception e) {
+        // log at trace level as closing an aborted stream will throw a content length
+        // check exception with the Apache HTTP client
+        LOG.trace("Error closing stream", e);
+      }
+      stream = null;
+    }
+  }
+
+  private void abortStream() {
+    try {
+      if (stream instanceof Abortable && stream.read() != -1) {
+        ((Abortable) stream).abort();
+      }
+    } catch (Exception e) {
+      LOG.debug("An error occurred while aborting the stream", e);
```

Comment:
Is this expected? If not we should log with warn

Reply:
This shouldn't happen, I went ahead and changed this to `warn`. I also went ahead and removed the `trace` log for `close()`.